services/tables/src/main/java/com/linkedin/openhouse/tables/config/MainApplicationConfig.java

```diff
@@ -80,7 +84,17 @@ MeterRegistryCustomizer<MeterRegistry> provideMeterRegistry() {
   Consumer<Supplier<Path>> provideFileSecurer() {
     return pathSeqSupplier -> {
       try {
-        FsStorageUtils.securePath(fsStorageProvider.storageClient(), pathSeqSupplier.get());
+        // TODO: This should use high-level storage api such as Storage::secureTableObject.
```

Comment:
what about having secure-storage-space as a interface somewhere? 
this code block seems a bit awkward when there's third option coming in.

Reply:
But I don't find a good place for this with your current design. nothing under `com.linkedin.openhouse.cluster.storage` seems to be fit.

Reply:
Yes, we'd need to address security aspect eventually and thoughtfully, that's why intentionally kept out of PR.
Curious to know why `com.linkedin.openhouse.cluster.storage` wouldn't be a good fit

Reply:
I was referring to existing classes within `com.linkedin.openhouse.cluster.storage` - The other thread you proposed `Storage` but isn't that too high-level?

Reply:
I want any high-level expectations of OH to the storage to be in the `Storage` class, example could be: `Storage::allocateTableObjectSpace` , `Storage::doesTableObjectExist`